src/TableFunctions/TableFunctionAzureBlobStorage.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AZURE_BLOB_STORAGE
+
+#include <TableFunctions/ITableFunction.h>
+#include <Storages/StorageAzureBlob.h>
+
+
+namespace DB
+{
+
+class Context;
+
+/* AzureBlob(source, [access_key_id, secret_access_key,] [format, structure, compression]) - creates a temporary storage for a file in AzureBlob.
+ */
+class TableFunctionAzureBlobStorage : public ITableFunction
+{
+public:
+    static constexpr auto name = "azure_blob_storage";
+    static constexpr auto signature = "- connection_string|storage_account_url, container_name, blobpath, [account_name, account_key, format, compression, structure]\n";
+
+    static size_t getMaxNumberOfArguments() { return 8; }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    virtual String getSignature() const
+    {
+        return signature;
+    }
+
+    bool hasStaticStructure() const override { return configuration.structure != "auto"; }
+
+    bool needStructureHint() const override { return configuration.structure == "auto"; }
+
+    void setStructureHint(const ColumnsDescription & structure_hint_) override { structure_hint = structure_hint_; }
+
+    bool supportsReadingSubsetOfColumns() override;
+
+    std::unordered_set<String> getVirtualsToCheckBeforeUsingStructureHint() const override
+    {
+        return {"_path", "_file"};
+    }
+
+    static StorageAzureBlob::Configuration parseArgumentsImpl(ASTs & args, const ContextPtr & context, bool get_format_from_file = true);
```

Comment:
```suggestion
    void parseArgumentsImpl(ASTs & args, const ContextPtr & context);
```

@alesapin  `get_format_from_file` is not used (or did you add it for future) and we can remove static as its updates the `configuration` of the class.